llvm/test/CodeGen/DirectX/scalar-bug-117273.ll

```diff
@@ -8,12 +8,18 @@
 define internal void @main() #1 {
 ; CHECK-LABEL: define internal void @main() {
 ; CHECK-NEXT:  [[ENTRY:.*:]]
-; CHECK-NEXT:    [[DOTI0:%.*]] = load float, ptr getelementptr inbounds ([24 x float], ptr @StaticArr.scalarized.1dim, i32 1), align 16
```

Comment:
This was an unexpected test update, Both the before and after both look right.

This behavior seems to have something to do with  the fact that we aren't walking uses and doing `cast<ConstantExpr>(U);` anymore. Instrad walking instructons in  basic blocks via `for (Instruction &I : make_early_inc_range(*BB))` seems to force the ConstantExpr to instructions so updated these tests.

Reply:
This is a little surprising but it seems fine. Do we also have tests where the GEP was a constant expression in the first place to make sure that that's still doing the right thing?